_sass/blog.scss

```diff
@@ -139,6 +139,40 @@
   }
 }
 
+@media screen and (max-width: 520px) {
+
+  .blog-post .post-details {
+    font-size: 0.68em;
+  }
+  .blog-filter a {
+    font-size: 0.68em;
+  }
+  
+  .blog-post h3 a {
+    font-size: 1em;
+  }
+  .page h2{
+    font-size: 1.5em;
+  }
+
+}
+
+@media screen and (max-width: 370px) {
+
+  .blog-post .post-details {
+    font-size: 0.6em;
+  }
+  .blog-filter a {
+    font-size: 0.6em;
+  }
+  
+  .blog-post h3 a {
+    font-size: 1em;
+  }
+  .page h2{
+    font-size: 1.5em;
+  }
+}
```

Comment:
Instead try using `flex-column` and `flex-direction: column-reverse` for  `.blog-post .post-details` and `align-items: flex-start` for ` .post-date`  at smaller screens.